### PR TITLE
Sanitise storageclasses

### PR DESCRIPTION
**Pull request by scrungus**
We need to sanitise storageclass resource names to avoid:
```
status:
  failureMessage: "Error: 2 errors occurred:\n\t* StorageClass.storage.k8s.io \"Ceph-HDD\"
    is invalid: metadata.name: Invalid value: \"Ceph-HDD\": a lowercase RFC 1123 subdomain
    must consist of lower case alphanumeric characters, '-' or '.', and must start
    and end with an alphanumeric character (e.g. 'example.com', regex used for validation
    is '[a-z0-9]([-a-z0-9]*[a-z0-9])?(\\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*')\n\t* StorageClass.storage.k8s.io
    \"__DEFAULT__\" is invalid: metadata.name: Invalid value: \"__DEFAULT__\": a lowercase
    RFC 1123 subdomain must consist of lower case alphanumeric characters, '-' or
    '.', and must start and end with an alphanumeric character (e.g. 'example.com',
    regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?(\\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*')\n\n\n"
  phase: Failed
```
on `csi-cinder-storageclass` manifests

### CODE REVIEW

File: magnum_capi_helm/driver_utils.py [hunk 1]
```diff
@@ -34,7 +34,7 @@ def sanitized_name(name, suffix=None):
         "[^a-z0-9]+",
         "-",
         (f"{name}-{suffix}" if suffix else name).lower(),
-    )
+    ).strip("-")
```
Comment (JohnGarbutt) [inline magnum_capi_helm/driver_utils.py:37]: Why is this needed?

File: magnum_capi_helm/driver.py [hunk 2]
```diff
@@ -691,7 +691,7 @@ def _storageclass_definitions(self, context, cluster):
 
         for volume_type in volume_types:
             storage_class = {
-                "name": volume_type,
+                "name": driver_utils.sanitized_name(volume_type),
```
Comment (JohnGarbutt) [inline magnum_capi_helm/driver.py:694]: Can you tweak a test to make sure this does the correct thing please?

**Review by JohnGarbutt — DISMISSED** [review]

File: magnum_capi_helm/driver_utils.py (hunk 1 above)
Comment (scrungus) [reply magnum_capi_helm/driver_utils.py:37]: subdomains have to start and end with alphanumerics, so a name like `__DEFAULT__` would be `--default--` without this

File: magnum_capi_helm/driver.py (hunk 2 above)
Comment (scrungus) [reply magnum_capi_helm/driver.py:694]: yep, I tweaked the test, just forgot to push it 😅

**Review by mkjpryor — APPROVED** [review]
LGTM
